src/snowflake/snowpark/mock/plan.py

```diff
@@ -1373,7 +1399,15 @@ def calculate_expression(
                     else:
                         res_cols.append(None)
             res_col = ColumnEmulator(
-                data=res_cols, dtype=object
+                data=res_cols,
+                dtype=object,
+                sf_type=calculate_expression(
+                    window_function.expr,
+                    windows[0],
```

Comment:
We could use `windows[0]` but I also think we could use input_data directly. All the required columns should be present.

Reply:
updated